Scripts/UI/GradientButton.cs

```diff
@@ -92,23 +110,57 @@ public bool visible
 
 				m_Visible = value;
 
+				if (m_Visible && !gameObject.activeSelf)
+					gameObject.SetActive(true);
+
 				this.StopCoroutine(ref m_VisibilityCoroutine);
 				m_VisibilityCoroutine = value ? StartCoroutine(AnimateShow()) : StartCoroutine(AnimateHide());
 			}
 		}
 		bool m_Visible;
 
+		Vector3 originalScale
+		{
+			get { return invertHighlightScale ? m_HighlightContentContainerLocalScale : m_OriginalContentContainerLocalScale; }
+		}
+
+		Vector3 highlightScale
+		{
+			get { return invertHighlightScale ? m_OriginalContentContainerLocalScale : m_HighlightContentContainerLocalScale; }
+		}
+
+		public Sprite icon
+		{
+			set
+			{
+				m_Icon.sprite = value;
+				m_OriginalIconSprite = value;
+			}
+		}
+
+		public bool invertHighlightScale { private get; set; }
+		public float containerContentsAnimationSpeedMultiplier { set { m_ContainerContentsAnimationSpeedMultiplier = value; }}
```

Comment:
It seems this is only ever set to a const value. Why not just set the individual durations in the prefab?

Reply:
Good suggestion, I exposed m_ContainerContentsAnimationSpeedMultiplier in GradientButton, and set it in the PinnedToolButton prefab.